pkg/virt-controller/watch/network_test.go

```diff
@@ -261,6 +262,81 @@ var _ = Describe("Network interface hot{un}plug", func() {
 			[]v1.VirtualMachineInterfaceRequest{},
 		),
 	)
+
+	Context("handle dynamic interface requests", func() {
+		var clusterConfig *virtconfig.ClusterConfig
+
+		BeforeEach(func() {
+			clusterConfig, _, _ = testutils.NewFakeClusterConfigUsingKVConfig(&v1.KubeVirtConfiguration{})
+		})
+
+		DescribeTable("for a VM with no interfaces",
+			func(requests []v1.VirtualMachineInterfaceRequest, expectedInterfaces []v1.Interface, vmi *v1.VirtualMachineInstance) {
+				vm := kubecli.NewMinimalVM("test")
+				vm.Spec.Template = &v1.VirtualMachineInstanceTemplateSpec{}
+				vm.Status.InterfaceRequests = requests
+
+				Expect(handleDynamicInterfaceRequests(vm, vmi, clusterConfig)).To(Succeed())
+				Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces).To(Equal(expectedInterfaces))
+			},
+			Entry("no request",
+				[]v1.VirtualMachineInterfaceRequest{},
+				nil,
+				vmiWithDefaultNet(),
+			),
+			Entry("one hot plug request, a default interface should be added to the VM",
+				[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
+				[]v1.Interface{
+					{Name: "default", InterfaceBindingMethod: v1.InterfaceBindingMethod{Bridge: &v1.InterfaceBridge{}}},
+					{Name: "blue", InterfaceBindingMethod: v1.InterfaceBindingMethod{Bridge: &v1.InterfaceBridge{}}},
+				},
+				vmiWithDefaultNet(),
+			),
+			Entry("one hot plug request, VMI has no pod network",
+				[]v1.VirtualMachineInterfaceRequest{{AddInterfaceOptions: &v1.AddInterfaceOptions{Name: "blue"}}},
+				[]v1.Interface{
+					{Name: "blue", InterfaceBindingMethod: v1.InterfaceBindingMethod{Bridge: &v1.InterfaceBridge{}}},
+				},
+				libvmi.New(),
+			),
+			Entry("one hot unplug request",
+				[]v1.VirtualMachineInterfaceRequest{{RemoveInterfaceOptions: &v1.RemoveInterfaceOptions{Name: "blue"}}},
+				nil,
+				vmiWithDefaultNet(),
+			),
+		)
+
+		DescribeTable("for a VM with existing interfaces",
+			func(requests []v1.VirtualMachineInterfaceRequest, expectedInterfaces []v1.Interface) {
+				vm := kubecli.NewMinimalVM("test")
+				vm.Spec.Template = &v1.VirtualMachineInstanceTemplateSpec{}
+				vm.Spec.Template.Spec.Domain.Devices.Interfaces = []v1.Interface{{Name: "oldNet", InterfaceBindingMethod: v1.InterfaceBindingMethod{Bridge: &v1.InterfaceBridge{}}}}
+				vm.Status.InterfaceRequests = requests
+
+				Expect(handleDynamicInterfaceRequests(vm, vmiWithDefaultNet(), clusterConfig)).To(Succeed())
+				Expect(vm.Spec.Template.Spec.Domain.Devices.Interfaces).To(Equal(expectedInterfaces))
```

Comment:
For completion, I think we also need to check the networks.

Reply:
done.